homeassistant/components/devolo_home_network/sensor.py

```diff
@@ -93,6 +122,29 @@ async def async_setup_entry(
                 device,
             )
         )
+        network = await device.plcnet.async_get_network_overview()
+        peers = [
+            peer.mac_address for peer in network.devices if peer.topology == REMOTE
+        ]
```

Comment:
Why do we extract the mac address of the peer here, only to then lookup the peer based on the mac address in the `DevoloPlcDataRateSensorEntity` constructor? Can't we just pass the peer directly?

Reply:
Hi @emontnemery , the data structure is a bit unlucky for my usecase, but maybe you see a better approach:

```
{
  'network':  {
    'devices':  [
      {
        'product_name': 'devolo dLAN pro 1200+ WiFi ac',
        'product_id': 'MT2730',
        'friendly_version': '2.8.0.01',
        'full_version': 'MAC-QCA7500-2.8.0.30-01-20190705-CS',
        'user_device_name': '',
        'mac_address': 'AABBCCDDEEFF',
        'topology': 'LOCAL',
        'technology': 'HPAV_PANTHER',
        'bridged_devices': [],
        'user_network_name': '',
        'ipv4_address': '',
        'attached_to_router': False
      },
      {
        'product_name': 'devolo dLAN 1200+',
        'product_id': 'MT2639',
        'friendly_version': '2.8.0.01-1',
        'full_version': 'MAC-QCA7500-2.8.0.30-01-20190705-CS',
        'user_device_name': '',
        'mac_address': '112233445566',
        'topology': 'REMOTE',
        'technology': 'HPAV_PANTHER',
        'bridged_devices': [],
        'attached_to_router': True,
        'user_network_name': '',
        'ipv4_address': ''
      }
    ],
    'data_rates':  [
      {
        'mac_address_from': 'AABBCCDDEEFF',
        'mac_address_to': '112233445566',
        'tx_rate': 129.9375,
        'rx_rate': 124.6875
      },
      {
        'mac_address_from': '112233445566',
        'mac_address_to': 'AABBCCDDEEFF',
        'tx_rate': 124.6875,
        'rx_rate': 129.9375
      }
    ]
  }
}
```

So what I basically do is, I grep the MAC of all devices that are REMOTE and then look which data rates match the device I'm looking at (the local device) and the peer device. In case I have three devices and the local device would be A, I would see six pairs of data rates (A-B, A-C, B-A, B-C, C-A, C-B) although I'm only interested in two of them (A-B, A-C).